File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeReadyMessage.scala





















// Generated on 05/03/2013 10:34:55
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeReadyMessage extends MessageDeserializer {
  val messageId = 5511

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeReadyMessage.messageId

    val ready: Boolean
    val step: Short

    def serialize(buf: Buffer) {
      buf.writeBoolean(ready)
      buf.writeShort(step)
    }

  }

  protected def create(ready: Boolean, step: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val ready = buf.readBoolean()
    val step = buf.readShort()

    create(ready, step)
  }

}

object ExchangeReadyMessage extends ExchangeReadyMessage {
  case class MessageImpl(ready: Boolean, step: Short) extends Message
  type Target = MessageImpl

  override def create(ready: Boolean, step: Short) = MessageImpl(ready, step)

  def apply(ready: Boolean, step: Short) = create(ready, step)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeReadyMessage.java


// Generated on 05/08/2013 19:37:54
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeReadyMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5511;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public boolean ready;
    public short step;
    
    public ExchangeReadyMessage() { }
    
    public ExchangeReadyMessage(boolean ready, short step) {
        this.ready = ready;
        this.step = step;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeBoolean(ready);
        buf.writeShort(step);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        ready = buf.readBoolean();
        step = buf.readShort();
        if (step < 0)
            throw new RuntimeException("Forbidden value on step = " + step + ", it doesn't respect the following condition : step < 0");
    }
    
}
